apps/heft/src/stages/BuildStage.ts

```diff
@@ -307,7 +307,8 @@ export class BuildStage extends StageBase<BuildStageHooks, IBuildStageProperties
     buildStage,
     watchMode
   }: IRunSubstageWithLoggingOptions): Promise<void> {
-    if (buildStage.hooks.run.isUsed()) {
+    // isUsed() also checks interceptors
+    if (buildStage.hooks.run.taps.length) {
```

Comment:
Let's make this more explicit

```suggestion
    if (buildStage.hooks.run.taps.length > 0) {
```